### ISSUE TITLE
Buildings under construction on a planet gifted to an ally are invisible to the receiver

### ISSUE THREAD

**Issue by Oberlus** [freeorion/freeorion#2677]
### Environment
* **FreeOrion Version**: v0.4.9 2019-12-15
* **Operating System**: Windows
  * Weekly development build

### Description
Buildings under construction (not completed) on a planet that is gifted to an ally are invisible to the receiving empire and get stuck in the queue. This causes a series of problems, for example:
If the queued building is a shipyard (maximum one per location), the receiving empire won't be able to queue one of its own, finish the one that is already there (but invisible) or remove it from the queue. See screenshots below:
![GiftedPlanetInvisUnderCons_ReceiverView](https://user-images.githubusercontent.com/28650914/71189001-5aa3d280-2282-11ea-80c8-b272530c2597.png)
The receiver empire is able to enqueue Geointegration Facility despite no shipyard present on the planet or enqueued, although the Geo. Fac. will never finish.

![GiftedPlanetInvisUnderCons_GifterView](https://user-images.githubusercontent.com/28650914/71189002-5aa3d280-2282-11ea-8a12-6e7e00fa8b43.png)
The gifter empire still has those buildings on its own queue for that planet that no longer owns.

If the gifter removes from its queue the offending buildings, the receiver empire will be able to operate normally on following turn.

I assume the same happens with unfinished batchs of ships.

### Expected Result
When gifting a planet to an ally, the buildings and ships under construction with its current state are acquired by the receiving empire (become part of its queue, attached at the end), and disapear from the gifter empire. If the receiver empire does not meet the requirements for any item in the queue, it should be shown as due "never", or if this is not supported be removed from the queue.

### Notes
I don't know if the same happens with conquered planets. I don't recall having problems of the kind of not being able to build something, but it's worth testing.
If the same problem happens, I'm unsure would would be better:
- Completely remove buildings and ships under construction when conquered.
- Or the same as when gifting: the conqueror acquires the incomplete projects.

//Edit:
In fact, I'm sure I've conquered (AI) planets that had some incompleted shipyards organic upgrades that ended up in my queue to either remove or complete if meeting the requirements.
That's exactly what I expect from the gifting mechanic.
I guess the codes for conquering and gifting could be compared to help fix the issue
//

### Steps to reproduce
Start FreeOrion and host a multiplayer game as Oberlus, load attached save game.
Start 2 more clients and join multiplayer localhost as Oberlus2 and Oberlus3.
Make blue empire gift Halley II planet to green empire and press turn, issue happens.
[FreeOrion_Oberlus_Oberlus_0068_20191219_163702.zip](https://github.com/freeorion/freeorion/files/3984544/FreeOrion_Oberlus_Oberlus_0068_20191219_163702.zip)

**Comment by Vezzra** [comment]
That sounds potentially serious. Should be fixed for 0.4.9.

**Comment by geoffthemedio** [comment]
Try
````
 server/ServerApp.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/server/ServerApp.cpp b/server/ServerApp.cpp
index b68d3712d..74fa6be74 100644
--- a/server/ServerApp.cpp
+++ b/server/ServerApp.cpp
@@ -3107,6 +3107,8 @@ namespace {
                         contained_obj->SetOwner(recipient_empire_id);
                 }
                 gifted_obj->SetOwner(recipient_empire_id);
+
+                Empire::ConquerProductionQueueItemsAtLocation(gifted_obj->ID(), recipient_empire_id);
             }
         }
     }
````
That should "conquer" the production items at the gifted location, to the recipient empire.

Edit: Tested, made a pull request. Seems to work...

![gifted_planet_prodution_items](https://user-images.githubusercontent.com/11386908/71242005-36d99e80-230d-11ea-850c-34f1e5b1a5d2.png)
